cockatrice/src/releasechannel.cpp

```diff
@@ -0,0 +1,312 @@
+        return;
+    }
+
+    if(!lastRelease)
+    	lastRelease = new Release;
+
+	lastRelease->setName(resultMap["name"].toString());
+	lastRelease->setDescriptionUrl(resultMap["html_url"].toString());
+	lastRelease->setPublishDate(resultMap["published_at"].toDate());
+
+    qDebug() << "Got reply from release server, size=" << tmp.size()
+        << "name=" << lastRelease->getName()
+        << "desc=" << lastRelease->getDescriptionUrl()
+        << "date=" << lastRelease->getPublishDate();
+
+    QString url = QString(STABLETAG_URL) + resultMap["tag_name"].toString();
+    qDebug() << "Searching for a corresponding tag on the stable channel: " << url;
+    response = netMan->get(QNetworkRequest(url));
+    connect(response, SIGNAL(finished()), this, SLOT(tagListFinished()));
+}
+
+void StableReleaseChannel::tagListFinished()
+{
+    QNetworkReply *reply = static_cast<QNetworkReply *>(sender());
+    bool ok;
+    QString tmp = QString(reply->readAll());
+    reply->deleteLater();
+
+    QVariantMap resultMap = QtJson::Json::parse(tmp, ok).toMap();
+    if (!ok) {
+        qWarning() << "No reply received from the tag update server:" << tmp;
+        emit error(tr("No reply received from the tag update server."));
+        return;
+    }
+
+    if(!(resultMap.contains("object") && 
+        resultMap["object"].toMap().contains("sha"))) {
+        qWarning() << "Invalid received from the tag update server:" << tmp;
+        emit error(tr("Invalid reply received from the tag update server."));
+        return;
+    }
+
+    lastRelease->setCommitHash(resultMap["object"].toMap()["sha"].toString());
+    qDebug() << "Got reply from tag server, size=" << tmp.size()
+        << "commit=" << lastRelease->getCommitHash();
+
+    qDebug() << "Searching for a corresponding file on the stable channel: " << QString(STABLEFILES_URL);
+    response = netMan->get(QNetworkRequest(QString(STABLEFILES_URL)));
+    connect(response, SIGNAL(finished()), this, SLOT(fileListFinished()));
+}
+
+void StableReleaseChannel::fileListFinished()
+{
+    QNetworkReply *reply = static_cast<QNetworkReply *>(sender());
+    bool ok;
+    QString tmp = QString(reply->readAll());
+    reply->deleteLater();
+
+    QVariantList resultList = QtJson::Json::parse(tmp, ok).toList();
+    if (!ok) {
+        qWarning() << "No reply received from the file update server:" << tmp;
+        emit error(tr("No reply received from the file update server."));
+        return;
+    }
+
+    QString shortHash = lastRelease->getCommitHash().left(GIT_SHORT_HASH_LEN);
+    QString myHash = QString(VERSION_COMMIT);
+    qDebug() << "Current hash=" << myHash << "update hash=" << shortHash;
+
+    bool needToUpdate = (QString::compare(shortHash, myHash, Qt::CaseInsensitive) != 0);
+    bool compatibleVersion = false;
+
+    foreach(QVariant file, resultList)
+    {
+        QVariantMap map = file.toMap();
+        // TODO: map github version to bintray version
```

Comment:
Publish a new revision to make cockatrice "find" the update.
Then, it will always download the "latest" stable release available on bintray.